_security-analytics/api-tools/detector-api.md

```diff
@@ -308,21 +325,38 @@ PUT /_plugins/_security_analytics/detectors/J1RX1IMByX0LvTiGTddR
 }
 ```
 
+#### Response fields
```

Comment:
In the response for update detector we get the complete detector back. Sample response:
<img width="743" alt="image" src="https://user-images.githubusercontent.com/114732919/211864572-b95b6189-3919-47ed-bc95-584c55d77e7c.png">

Reply:
I noticed we have `Response fields` section only for the Update API, should we add it to other APIs as well?

Reply:
@amsiglan Since many of the meaningful fields are described in "Create Detector API", I added only fields found in responses that are novel and not described elsewhere. The three response fields I added for the "Update Detector API" looked like they hadn't been addressed, so I added them. I also posed this question to @sbcd90 for a number of "Search Detector API" response fields. The answer I got was that many of the fields for that API's response were either general OpenSearch search response fields, fields already covered in Create Detector API tables, or hidden fields that could be ignored. 
Still, are there specific fields in responses that you think are not covered? I tried to review the responses for missing fields. But I may have overlooked some.

Reply:
@amsiglan Actually, I'm not sure the Update Detector API sample request is accurate. I'm now getting this for a response when using the sample in documentation.
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "null_pointer_exception",
        "reason" : "Cannot invoke \"String.hashCode()\" because \"<local6>\" is null"
      }
    ],
    "type" : "null_pointer_exception",
    "reason" : "Cannot invoke \"String.hashCode()\" because \"<local6>\" is null"
  },
  "status" : 500
}
```
What does your request look like for the response above?

Reply:
This is what the request payload looks like:

```
{
  "type": "detector",
  "name": "test windows updated",
  "detector_type": "WINDOWS",
  "enabled": true,
  "enabled_time": 1673369808517,
  "schedule": {
    "period": {
      "interval": 1,
      "unit": "MINUTES"
    }
  },
  "inputs": [
    {
      "detector_input": {
        "description": "",
        "indices": [
          "windows"
        ],
        "custom_rules": [],
        "pre_packaged_rules": [
          {
            "id": "73a883d0-0348-4be4-a8d8-51031c2564f8"
          },
          {
            "id": "1a4bd6e3-4c6e-405d-a9a3-53a116e341d4"
          }
        ]
      }
    }
  ],
  "triggers": [
    {
      "id": "K1ScnIUBJUlf_Nvv7gSF",
      "name": "Test trigger",
      "severity": "1",
      "types": [
        "windows"
      ],
      "ids": [],
      "sev_levels": [
        "low"
      ],
      "tags": [],
      "actions": [
        {
          "id": "",
          "name": "Triggered alert condition:  - Severity: 1 (Highest) - Threat detector: test windows",
          "destination_id": "",
          "message_template": {
            "source": "Triggered alert condition: \nSeverity: 1 (Highest)\nThreat detector: test windows\nDescription: \nDetector data sources:\n\twindows",
            "lang": "mustache"
          },
          "throttle_enabled": false,
          "subject_template": {
            "source": "Triggered alert condition:  - Severity: 1 (Highest) - Threat detector: test windows",
            "lang": "mustache"
          },
          "throttle": {
            "value": 10,
            "unit": "MINUTES"
          }
        }
      ]
    }
  ],
  "last_update_time": 1674784886451,
  "monitor_id": [
    "PFScnIUBJUlf_Nvv8gRx"
  ],
  "bucket_monitor_id_rule_id": {
    "-1": "PFScnIUBJUlf_Nvv8gRx"
  },
  "rule_topic_index": ".opensearch-sap-windows-detectors-queries",
  "alert_index": ".opensearch-sap-windows-alerts",
  "alert_history_index": ".opensearch-sap-windows-alerts-history",
  "alert_history_index_pattern": "<.opensearch-sap-windows-alerts-history-{now/d}-1>",
  "findings_index": ".opensearch-sap-windows-findings",
  "findings_index_pattern": "<.opensearch-sap-windows-findings-{now/d}-1>"
}
```